### ISSUE TITLE
React 16 - Popup mode is not working

### ISSUE THREAD

**Issue by ucze** [ni3galave/react-x-editable#4]
Hi i would like to inform you that popup mode is not working with react 16. 

> bundle.js:8275 Uncaught Error: _processChildContext is not available in React 16+. This likely means you have multiple copies of React and are attempting to nest a React 15 tree inside a React 16 tree using unstable_renderSubtreeIntoContainer, which isn't supported. Try to make sure you have only one copy of React (and ideally, switch to ReactDOM.createPortal).

**Comment by ni3galave** [comment]
Thanks for raising the issue. I will look into it asap.

**Comment by ucze** [comment]
Hi, what is the status of this issue? :)

**Comment by ni3galave** [comment]
React v16 versioning issue is fixed : https://github.com/ni3galave/react-x-editable/commit/d934006b0109ad9262a3d4f7b94103c6e699625e

**Comment by ni3galave** [comment]
[Added support for React 16+. ](https://github.com/ni3galave/react-x-editable/releases/tag/v0.0.10-beta)

Please let me know if you face issue.

**Comment by cpratiksha** [comment]
Popup issue is working on my local, how to call this function inside another tag ?
<td>
              <InputStyle
                handleChange={this.handleChange}
                displayValue={this.state.tag}
                name="tag"
                underline
                style={{ textDecorationColor: '#0997DE' }}
                set={this.state.set}
                // uid={this.state.uid}
                saving={this.props.saving}
                updateSave={this.props.updateSave}
                href="javascript:;"
              />
        ...working this editable tag but ,i want to call this function on inputStyle tag 
              <Editable name="username" dataType="text" mode="popup" />
            </td>

**Comment by cpratiksha** [comment]
<td>
              <InputStyle
                handleChange={this.handleChange}
                displayValue={this.state.tag}
                name="tag"
                underline
                style={{ textDecorationColor: '#0997DE' }}
                set={this.state.set}
                // uid={this.state.uid}
                saving={this.props.saving}
                updateSave={this.props.updateSave}
                href="javascript:;"
              />
              <Editable name="username" dataType="text" mode="popup" />
            </td>

**Comment by YassienW** [comment]
After doing a little digging, i discovered that the popups on the example website (and in my project) aren't working properly because Bootstrap 4 is being used instead of Bootstrap 3. If you inspect the html the popups are added to the DOM but they simply don't show. This library depends on [react-bootstrap](https://react-bootstrap.github.io/), which sadly doesn't have Bootstrap 4 support yet. I'll be trying to get it working with Bootstrap 4 since i already have a project using react-bootstrap & Bootstrap 4 without any issues

**Comment by YassienW** [comment]
Invisible popover issue fixed on [my fork](https://github.com/YassienW/react-x-editable/tree/dev) by using [reactstrap](https://reactstrap.github.io/) (not react-bootstrap) for Bootstrap 4 popovers. You will need to specify a `bootstrap4={true}` prop on the editable

**Comment by fcpauldiaz** [comment]
@YassienW would you do a PR of your fork to merge support for bs4?

**Comment by YassienW** [comment]
@fcpauldiaz I don't think the PR will get merged, the project looks abandoned to me. I did start my own similar project though but it doesn't have popover mode yet

**Comment by fcpauldiaz** [comment]
@YassienW  Hi 

How did you builded/compiled this project ?

I have this error when I try
```
ERROR in ./libs/js/Editable.jsx
Module parse failed: /react-x-editable/libs/js/Editable.jsx Unexpected token (62:18)
You may need an appropriate loader to handle this file type.
|     }
|   }
|   setInitialValue = (newProps) => {
|     const { dataType, options, value } = newProps;
|     if(dataType == "select" || dataType == "checklist"){
```

**Comment by YassienW** [comment]
I just ran the scripts in package.json, they work fine for me.

I did implement popover mode in [my own project](https://github.com/YassienW/react-bootstrap-editable), if you would like to try it out. Feedback would be appreciated.

**Comment by fcpauldiaz** [comment]
Nice work @YassienW. I noticed that it uses font-awesome but I didn't found any dependency to font-awesome, it should be loaded in the parent project using it ? (https://github.com/YassienW/react-bootstrap-editable/blob/master/src/Editable.js#L40)

P.S. I would like to be a contributor

**Comment by YassienW** [comment]
@fcpauldiaz Yes, i should probably specify that. I think we should move any discussion related to my project to its repository to avoid spamming here.
